src/Content/Text.php

```diff
@@ -0,0 +1,1714 @@
+            } else {
+                $table = $vowels;
+            }
+
+        }
+
+        $word = substr($word,0,$len);
+
+        foreach ($noend as $noe) {
+            $noelen = strlen($noe);
+            if ((strlen($word) > $noelen) && (substr($word, -$noelen) == $noe)) {
+                $word = autoName($len);
+                break;
+            }
+        }
+
+        return $word;
+    }
+
+    /**
+     * escape text ($str) for XML transport
+     * @param string $str
+     * @return string Escaped text.
+     */
+    function xmlify($str) {
+        $buffer = htmlspecialchars($str, ENT_QUOTES, "UTF-8");
+        $buffer = trim($buffer);
+
+        return $buffer;
+    }
+
+    /**
+     * undo an xmlify
+     * @param string $s xml escaped text
+     * @return string unescaped text
+     */
+    function unxmlify($s) {
+        $ret = htmlspecialchars_decode($s, ENT_QUOTES);
+        return $ret;
+    }
+
+    /**
+     * Loader for infinite scrolling
+     * @return string html for loader
+     */
+    function scrollLoader() {
+        $tpl = self::getMarkupTemplate("scroll_loader.tpl");
+        return self::replaceMacros($tpl, [
+            'wait' => L10n::t('Loading more entries...'),
+            'end' => L10n::t('The end')
+        ]);
+    }
+
+    /**
+     * Turn user/group ACLs stored as angle bracketed text into arrays
+     *
+     * @param string $s
+     * @return array
+     */
+    function expandAcl($s) {
+        // turn string array of angle-bracketed elements into numeric array
+        // e.g. "<1><2><3>" => array(1,2,3);
+        $ret = [];
+
+        if (strlen($s)) {
+            $t = str_replace('<', '', $s);
+            $a = explode('>', $t);
+            foreach ($a as $aa) {
+                if (intval($aa)) {
+                    $ret[] = intval($aa);
+                }
+            }
+        }
+        return $ret;
+    }
+
+    /**
+     * Wrap ACL elements in angle brackets for storage
+     * @param string $item
+     */
+    function sanitiseAcl(&$item) {
+        if (intval($item)) {
+            $item = '<' . intval(self::noTags(trim($item))) . '>';
+        } else {
+            unset($item);
+        }
+    }
+
+
+    /**
+     * Convert an ACL array to a storable string
+     *
+     * Normally ACL permissions will be an array.
+     * We'll also allow a comma-separated string.
+     *
+     * @param string|array $p
+     * @return string
+     */
+    function perms2Str($p) {
+        $ret = '';
+        if (is_array($p)) {
+            $tmp = $p;
+        } else {
+            $tmp = explode(',', $p);
+        }
+
+        if (is_array($tmp)) {
+            array_walk($tmp, 'self::sanitiseAcl');
+            $ret = implode('', $tmp);
+        }
+        return $ret;
+    }
+
+    /**
+     * load template $s
+     *
+     * @param string $s
+     * @param string $root
+     * @return string
+     */
+    function getMarkupTemplate($s, $root = '') {
```

Comment:
Code standards: Please add a new line before class method opening brace.

Reply:
Additionally, this method is heavily coupled with the `App` object, I think it makes more sense to move it to the `App` class.